app/models/user.rb

```diff
@@ -26,7 +26,7 @@ class User < ActiveRecord::Base
   validates_uniqueness_of :name # done by devise
   validates_uniqueness_of :email # done by devise
 
-  validates :joinedwithcode, :presence => true, :inclusion => { :in => $codes, :message => "%{value} is not valid" }, :on => :create
+  validates :joinedwithcode, :presence => true, :inclusion => { :in => User.all.pluck(:code), :message => "%{value} is not valid" }, :on => :create
```

Comment:
OK solid I'll switch it back to $codes. I don't remember why I switched it, probably just because I didn't understand it.